.circleci/config.yml

```diff
@@ -1,37 +1,200 @@
+            tar -cvf artifact.tar *
+            mv artifact.tar ${OLDPWD}/
+      - run:
+          name: "Tag commit id as artifact identifer"
+          command: echo "${CIRCLE_SHA1}" > artifact-info.txt
+  upload_artifact:
+    description: "upload artifact to s3"
+    steps:
+      - s3/copy:
+          from: artifact.tar
+          to: 's3://${CONTEXT_ARTIFACT_S3_BUCKET}/${CIRCLE_PROJECT_REPONAME}/'
+          aws-access-key-id: env_CONTEXT_ARTIFACT_S3_AWS_ACCESS_KEY_ID
+          aws-secret-access-key: env_CONTEXT_ARTIFACT_S3_AWS_SECRET_ACCESS_KEY
+          aws-region: env_CONTEXT_ARTIFACT_S3_AWS_REGION
+          arguments: '--metadata "{\"x-amz-artifact-id\": \"${CIRCLE_SHA1}\" }"'
+  upload_checksum:
+    description: "upload artifact commit id to s3"
+    steps:
+      - s3/copy:
+          from: artifact-info.txt
+          to: 's3://${CONTEXT_ARTIFACT_S3_BUCKET}/${CIRCLE_PROJECT_REPONAME}/'
+          aws-access-key-id: env_CONTEXT_ARTIFACT_S3_AWS_ACCESS_KEY_ID
+          aws-secret-access-key: env_CONTEXT_ARTIFACT_S3_AWS_SECRET_ACCESS_KEY
+          aws-region: env_CONTEXT_ARTIFACT_S3_AWS_REGION
+  docker:
+    description: "Build and Push image to docker hub"
+    parameters:
+      target:
+        type: string
+    steps:
+      - setup_remote_docker
+      - run:
+          name: Building docker image for << parameters.target >>
+          command: |
+            build_tag="${CIRCLE_SHA1}"
+            [ "<< parameters.target >>" == "beta" ] && build_tag="beta-${CIRCLE_SHA1}"
+            docker build -t ${DOCKHUB_ORGANISATION}/binary-static-bot:${build_tag} -t ${DOCKHUB_ORGANISATION}/binary-static-bot:latest .
+      - run:
+          name: Pushing Image to docker hub
+          command: |
+            build_tag="${CIRCLE_SHA1}"
+            [ "<< parameters.target >>" == "beta" ] && build_tag="beta-${CIRCLE_SHA1}"
+            echo $DOCKERHUB_PASSWORD | docker login -u $DOCKERHUB_USERNAME --password-stdin
+            docker push ${DOCKHUB_ORGANISATION}/binary-static-bot:${build_tag}
+            docker push ${DOCKHUB_ORGANISATION}/binary-static-bot:latest
+  k8s_deploy:
+    description: "Deploy to k8s cluster"
+    parameters:
+      target:
+        type: string
+    steps:
+      - k8s/install-kubectl
+      - run:
+          name: Deploying to k8s cluster for service binary-bot-beta
+          command: |
+            for SERVER_ID in {1..5}
+            do
+              KUBE_SERVER_REF="KUBE_SERVER_$SERVER_ID"
+              SERVICEACCOUNT_TOKEN_REF="SERVICEACCOUNT_TOKEN_$SERVER_ID"
+              CA_CRT_REF="CA_CRT_$SERVER_ID"
+              if [ ! -z "${!KUBE_SERVER_REF}" ]
+              then
+                echo "Deploying to cluster $SERVER_ID"
+                CA_CRT="${!CA_CRT_REF}"
+                KUBE_SERVER="${!KUBE_SERVER_REF}"
+                SERVICEACCOUNT_TOKEN="${!SERVICEACCOUNT_TOKEN_REF}"
+                build_tag="${CIRCLE_SHA1}"
+                deployment_target="bot-binary-com"
```

Comment:
again if you declare `build_tag` and `deployment_target` as a parameter and set the value from workflow based on `staging` and `production` things would be much easier.